src/ui/states/state-needs-payment-info.svelte

```diff
@@ -125,10 +128,12 @@
             <Button disabled={processing}>
               {#if processing}
                 <ProcessingAnimation />
+              {:else if productDetails.subscriptionOptions[purchaseOptionToUse.id].trial !== null}
```

Comment:
Hmm instead of having to pass the product to this view, we could try to downcast the `purchaseOption` to a `subscriptionOption`, like:
```suggestion
              {:else if (purchaseOptionToUse as SubscriptionOption).trial !== undefined}
```
That way, we don't need to pass in the `productDetails`. Wdyt?

Reply:
I was going back and forth on this. I think the disadvantage there is that we might have to change that back when we implement one-time purchases, so I think I would leave it as is. I am making the condition more robust by adding optional chaining though so this doesn't break in the future if any of the fields are null (eg. if you have a one-time product that doesn't have subscription options)

Reply:
Ok I think this is fine for now 👍